src/isolation.rs

```diff
@@ -0,0 +1,225 @@
+use std::{
+	collections::HashMap,
+	ffi::{CString, OsString},
+	fmt, fs,
+	fs::Permissions,
+	os::unix::{ffi::OsStrExt, fs::PermissionsExt},
+	path::PathBuf,
+};
+
+use tempfile::{Builder, TempDir};
+use uuid::Uuid;
+
+/// Creates a temporary directory.
+pub fn create_temp_dir() -> TempDir {
+	let dir = Builder::new()
+		.permissions(Permissions::from_mode(0o700))
+		.prefix(&Uuid::new_v4().to_string())
+		.suffix("-uhyve")
+		.tempdir()
+		.ok()
+		.unwrap_or_else(|| panic!("The temporary directory could not be created."));
+
+	let dir_permissions = dir.path().metadata().unwrap().permissions();
+	assert!(!dir_permissions.readonly());
+
+	dir
+}
+
+/// HashMap matching a path in the guest OS ([String]) a path in the host OS ([OsString]).
+///
+/// Using a list of parameters stored in a [Vec<String>], this function creates
+/// a HashMap that can match a path on the host operating system given a path on
+/// the guest operating system.
+///
+/// See [crate::hypercall::open] to see this in practice.
+pub struct UhyveFileMap {
+	files: HashMap<String, OsString>,
+}
+
+impl UhyveFileMap {
+	/// Creates a UhyveFileMap.
+	///
+	/// * `parameters` - A list of parameters with the format `./host_path.txt:guest.txt`
+	pub fn new(parameters: &Option<Vec<String>>) -> UhyveFileMap {
```

Comment:
Instead of an `Option`, you could always directly use a `Vec`. If the caller doesn't have a mapping, the `Vec` will simply have length 0.

Reply:
I did this initially to spare the function call if the UhyveFileMap was empty / not to be used. Will correct.